packages/framework/src/Form/Admin/Payment/PaymentEditFormType.php

```diff
@@ -37,40 +36,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('paymentData', PaymentFormType::class, [
                 'payment' => $paymentDetail !== null ? $paymentDetail->getPayment() : null,
             ])
-            ->add($this->getPricesBuilder($builder))
+            ->add('pricesByCurrencyId', PriceTableType::class, [
+                'currencies' => $this->currencyFacade->getAllIndexedById(),
```

Comment:
If you will have e-shop with more than 2 currencies and you will decide that you want to set prices for only 2 out of 4 currencies, you are not able to do that if currencies are responsibility of PriceTableType. For me PriceTableType should get all data that user want render and render it for him. 

This way the PricetableType will be blocked to only one usage.

Reply:
Everything you said might be right "if". 

If you have e-commerce application with more than two currencies and you will decide two set prices for only two currencies you will probably want to make this change in PriceTableTypeExtension in one place, not in x places.

Anyway, right now we want to display all currencies because every usage of this type gets them. Please update it as I suggested. And do not forget my second comment, please.

Reply:
I disagree with you on this. I dont see any advantage of making PriceTableType responsible for currencies. In my approach if i will like to set prices for one currency, i will just change method which gets the currencies in `PaymentFormType`

With your approach, i would need to create Extension with what? what it will do?
Then i will need to actually change the behavior of PriceTableType.

So I dont really see why you think that your approach is btter.

Reply:
@MattCzerner's way makes using `PriceTableType` more flexible - you can use it just for some currencies. Also the user has control over the used indices and is not forced to indexing by currency id.

@TomasLudvik's way would make using `PriceTableType` easier - you don't have to be dependent on `CurrencyFacade` yourself and you avoid passing the currencies.

I wouldn't mind either - each has some advantages and disadvantages. If I were to choose, I would go with Matt's - the main reason is that it's already implemented.